### PR TITLE
Implement APIs for CampaignRating

### PR DESCRIPTION
**Pull request by cungbac**
1. Create CampaignRating

Path: https://localhost:7066/api/v1/campaign-rating/create
Method: POST
Body:
```
{
    "ratingPoint": 3,
    "feedback": "amazing",
    "campaignId": "3fb27518-9767-469a-b624-addfa65ee86d"
}
```
Response
```
{
    "id": "31eb9201-c790-4df5-8c62-5fda6b4c91dc",
    "ratingPoint": 3,
    "feedback": "amazing",
    "campaignId": "3fb27518-9767-469a-b624-addfa65ee86d",
    "createdAt": "2024-11-10T08:34:48.0501699Z",
    "createdBy": "f5c2063e-9a14-4ab4-bce2-7f5117ad5505",
    "updatedAt": "2024-11-10T08:34:48.0501699Z",
    "updatedBy": "f5c2063e-9a14-4ab4-bce2-7f5117ad5505"
}
```
2. Get CampaignRating by Id
Path: https://localhost:7066/api/v1/campaign-rating/{campaignRatingId}
Method: GET
Body: None
Response
```
{
    "id": "c2aebe42-25e9-43fe-b4e3-be6a0081aba7",
    "ratingPoint": 3,
    "feedback": "very happy",
    "campaignId": "686e6e36-904a-4bda-936f-9ed90e752225",
    "createdAt": "2024-11-09T17:17:38.537343Z",
    "createdBy": "f5c2063e-9a14-4ab4-bce2-7f5117ad5505",
    "updatedAt": "2024-11-09T17:19:58.905795Z",
    "updatedBy": "f5c2063e-9a14-4ab4-bce2-7f5117ad5505"
}

```

3. Update CampaignRating
Path: https://localhost:7066/api/v1/campaign-rating/{campaignRatingId}
Method: PUT
Body:
```
{
    "id": "c2aebe42-25e9-43fe-b4e3-be6a0081aba7",
    "ratingPoint": 3,
    "feedback": "very happy"
}
```
Response:
```
{
    "id": "c2aebe42-25e9-43fe-b4e3-be6a0081aba7",
    "ratingPoint": 3,
    "feedback": "very happy",
    "campaignId": "686e6e36-904a-4bda-936f-9ed90e752225",
    "createdAt": "2024-11-09T17:17:38.537343Z",
    "createdBy": "f5c2063e-9a14-4ab4-bce2-7f5117ad5505",
    "updatedAt": "2024-11-10T08:35:45.1232983Z",
    "updatedBy": "f5c2063e-9a14-4ab4-bce2-7f5117ad5505"
}
```
4. Delete CampaignRating
Path: https://localhost:7066/api/v1/campaign-rating/{campaignRatingId}
Method: DELETE
Body:
 None
Response:
```
{
    "id": "31eb9201-c790-4df5-8c62-5fda6b4c91dc",
    "ratingPoint": 3,
    "feedback": "amazing",
    "campaignId": "3fb27518-9767-469a-b624-addfa65ee86d",
    "createdAt": "2024-11-10T08:34:48.050169Z",
    "createdBy": "f5c2063e-9a14-4ab4-bce2-7f5117ad5505",
    "deletedAt": "2024-11-10T08:56:51.9475975Z",
    "deletedBy": "f5c2063e-9a14-4ab4-bce2-7f5117ad5505",
    "reasonForDeletion": null
}
```

### CODE REVIEW

File: src/DonateHope.Core/ServiceContracts/CampaignRatingsServiceContracts/ICampaignRatingCreatingService.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+using DonateHope.Core.DTOs.CampaignContributionDTOs;
+using DonateHope.Core.DTOs.CampaignRatingDTOs;
+using FluentResults;
+
+namespace DonateHope.Core.ServiceContracts.CampaignRatingsServiceContracts;
+
+public interface ICampaignRatingCreatingService
```
Comment (AnNguyenLe) [inline src/DonateHope.Core/ServiceContracts/CampaignRatingsServiceContracts/ICampaignRatingCreatingService.cs]: I did a bit of research about naming services - for this case, people prefer: `ICampaignRatingCreateService` over `ICampaignRatingCreatingService`. Let's follow the community @cungbac

Comment (AnNguyenLe) [reply src/DonateHope.Core/ServiceContracts/CampaignRatingsServiceContracts/ICampaignRatingCreatingService.cs]: Also, apply the same to other service. Such as `RetrievalService` -> `RetrieveService`

File: src/DonateHope.Core/ServiceContracts/CampaignRatingsServiceContracts/ICampaignRatingDeletingService.cs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+using DonateHope.Core.DTOs.CampaignRatingDTOs;
+using FluentResults;
+
+namespace DonateHope.Core.ServiceContracts.CampaignRatingsServiceContracts;
+
+public interface ICampaignRatingDeletingService
+{
```
Comment (AnNguyenLe) [inline src/DonateHope.Core/ServiceContracts/CampaignRatingsServiceContracts/ICampaignRatingDeletingService.cs]: Same here: `DeletingService` -> `DeleteService`

File: src/DonateHope.Core/ServiceContracts/CampaignRatingsServiceContracts/ICampaignRatingRetrievalService.cs [hunk 3]
```diff
@@ -0,0 +1,9 @@
+using DonateHope.Core.DTOs.CampaignRatingDTOs;
+using FluentResults;
+
+namespace DonateHope.Core.ServiceContracts.CampaignRatingsServiceContracts;
+
+public interface ICampaignRatingRetrievalService
+{
```
Comment (AnNguyenLe) [inline src/DonateHope.Core/ServiceContracts/CampaignRatingsServiceContracts/ICampaignRatingRetrievalService.cs]: `RetrievalService` -> `RetrieveService`

File: src/DonateHope.Core/ServiceContracts/CampaignRatingsServiceContracts/ICampaignRatingUpdatingService.cs [hunk 4]
```diff
@@ -0,0 +1,10 @@
+using DonateHope.Core.DTOs.CampaignContributionDTOs;
+using DonateHope.Core.DTOs.CampaignRatingDTOs;
+using FluentResults;
+
+namespace DonateHope.Core.ServiceContracts.CampaignRatingsServiceContracts;
+
+public interface ICampaignRatingUpdatingService
+{
+    Task<Result<CampaignRatingGetResponseDto>> UpdateCampaignRatingAsync(CampaignRatingUpdateRequestDto updateRequestDto, Guid userId);
```
Comment (AnNguyenLe) [inline src/DonateHope.Core/ServiceContracts/CampaignRatingsServiceContracts/ICampaignRatingUpdatingService.cs]: `UpdatingService` -> `UpdateService`

File: src/DonateHope.Infrastructure/Repositories/CampaignRatingsRepository.cs [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+    public IQueryable<CampaignRating> GetCampaignRatings(Expression<Func<CampaignRating, bool>> predicate)
+    {
+        return _dbContext.CampaignRatings.Where(predicate);
+    }
+
+    public async Task<Result<CampaignRating>> GetCampaignRatingById(Guid campaignRatingId)
+    {
+        using var dbConnection = await _dbConnectionFactory.CreateConnectionAsync();
+        var queryResult = await _dbContext
+            .CampaignRatings.Where(cc => cc.Id == campaignRatingId)
```
Comment (AnNguyenLe) [inline src/DonateHope.Infrastructure/Repositories/CampaignRatingsRepository.cs]: Not much but let's change the name of local variable `cc` -> `cr`

File: src/DonateHope.WebAPI/StartupExtensions/DependencyInjectionServicesExtension.cs [hunk 6]
```diff
@@ -44,14 +47,23 @@ IConfiguration configuration
...
+        services.TryAddSingleton<CampaignRatingMapper>();
+        services.TryAddScoped<ICampaignRatingCreatingService, CampaignRatingCreatingService>();
+        services.TryAddScoped<ICampaignRatingRetrievalService, CampaignRatingRetrievalService>();
+        services.TryAddScoped<ICampaignRatingUpdatingService, CampaignRatingUpdatingService>();
+        services.TryAddScoped<ICampaignRatingDeletingService, CampaignRatingDeletingService>();
+        services.AddValidatorsFromAssemblyContaining<CampaignRatingUpdateRequestValidator>();
+        
         return services;
     }
 }
```
Comment (AnNguyenLe) [inline src/DonateHope.WebAPI/StartupExtensions/DependencyInjectionServicesExtension.cs:69]: Thank you for your hard work!

Comment (cungbac) [reply src/DonateHope.WebAPI/StartupExtensions/DependencyInjectionServicesExtension.cs:69]: refactored

**Pull request merged**